catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/RemoteSolrCatalogProvider.java

```diff
@@ -0,0 +1,303 @@
+        super.maskId(id);
+        provider.maskId(id);
+    }
+
+    /**
+     * Used to signal to the Solr client to commit on every transaction. Updates
+     * the underlying ConfigurationStore so that the property is propagated
+     * throughout the Solr Catalog Provider code
+     *
+     * @param forceAutoCommit
+     */
+    public void setForceAutoCommit(boolean forceAutoCommit) {
+        ConfigurationStore.getInstance()
+                .setForceAutoCommit(forceAutoCommit);
+    }
+
+    public void setDisableTextPath(boolean disableTextPath) {
+        ConfigurationStore.getInstance()
+                .setDisableTextPath(disableTextPath);
+    }
+
+    @Override
+    public Set<ContentType> getContentTypes() {
+        return getProvider().getContentTypes();
+    }
+
+    @Override
+    public boolean isAvailable() {
+        return getProvider().isAvailable();
+    }
+
+    @Override
+    public boolean isAvailable(SourceMonitor callback) {
+        return getProvider().isAvailable(callback);
+    }
+
+    @Override
+    public SourceResponse query(QueryRequest queryRequest) throws UnsupportedQueryException {
+        return getProvider().query(queryRequest);
+    }
+
+    @Override
+    public String getDescription() {
+        return describableProperties.getProperty("description", "");
+    }
+
+    @Override
+    public String getOrganization() {
+        return describableProperties.getProperty("organization", "");
+    }
+
+    @Override
+    public String getTitle() {
+        return describableProperties.getProperty("name", "");
+    }
+
+    @Override
+    public String getVersion() {
+        return describableProperties.getProperty("version", "");
+    }
+
+    @Override
+    public CreateResponse create(CreateRequest createRequest) throws IngestException {
+        return getProvider().create(createRequest);
+    }
+
+    @Override
+    public DeleteResponse delete(DeleteRequest deleteRequest) throws IngestException {
+        return getProvider().delete(deleteRequest);
+    }
+
+    @Override
+    public UpdateResponse update(UpdateRequest updateRequest) throws IngestException {
+        return getProvider().update(updateRequest);
+    }
+
+    public void shutdown() {
+        closeSolrClient();
+    }
+
+    /**
+     * Shutdown the connection to Solr and releases resources.
+     */
+    private void closeSolrClient() {
+        LOGGER.debug("Closing connection to Solr client.");
+        if (getClient() != null) {
+            try {
+                getClient().close();
+            } catch (IOException e) {
+                LOGGER.info("Unable to close Solr client", e);
+            }
+        } else if (!clientFuture.isDone() && !clientFuture.isCancelled()) {
+            clientFuture.cancel(true);
+        }
+        LOGGER.debug("Finished closing connection to Solr client.");
+    }
+
+    public String getUrl() {
+        return url;
+    }
+
+    public void setUrl(String url) {
```

Comment:
Should we add a JavaDoc comment here since this does more than just save off the URL?

Actually, are `getUrl` and `setUrl` even used?